src/functions/verb.js

```diff
@@ -0,0 +1,53 @@
+// @flow
+import defineFunction from "../defineFunction";
+import buildCommon from "../buildCommon";
+import mathMLTree from "../mathMLTree";
+import ParseError from "../ParseError";
+
+defineFunction({
+    type: "verb",
+    names: ["\\verb"],
+    props: {
+        numArgs: 0,
+        allowedInText: true,
+    },
+    handler(context, args, optArgs) {
+        // \verb and \verb* are dealt with directly in Parser.js.
+        // If we end up here, it's because of a failure to match the two delimiters
+        // in the regex in Lexer.js.  LaTeX raises the following error when \verb is
+        // terminated by end of line (or file).
+        throw new ParseError(
+            "\\verb ended by end of line instead of matching delimiter");
+    },
+    htmlBuilder(group, options) {
+        const text = buildCommon.makeVerb(group, options);
+        const body = [];
+        // \verb enters text mode and therefore is sized like \textstyle
+        const newOptions = options.havingStyle(options.style.text());
+        for (let i = 0; i < text.length; i++) {
+            if (text[i] === '\xA0') {  // spaces appear as nonbreaking space
+                // The space character isn't in the Typewriter-Regular font,
+                // so we implement it as a kern of the same size as a character.
+                // 0.525 is the width of a texttt character in LaTeX.
+                // It automatically gets scaled by the font size.
+                const rule = buildCommon.makeSpan(["mord", "rule"], [], newOptions);
+                rule.style.marginLeft = "0.525em";
+                body.push(rule);
+            } else {
+                body.push(buildCommon.makeSymbol(text[i], "Typewriter-Regular",
+                    group.mode, newOptions, ["mathtt"]));
+            }
+        }
+        buildCommon.tryCombineChars(body);
+        // $FlowFixMe: CombinableDomNode[] is not compatible with DomChildNode[]
```

Comment:
Where did `CombinableDomNode[]` get inferred by the type system? I don't see anything explicitly requiring that. Does declaring `body` above explicitly as `DomChildNode[]` resolve this?

Reply:
The problem is that `tryCombineChars` expects `CombinableDomNode[]` but the call to `makeSpan` below expects `DomChildNode[]` and the two aren't compatible.  I'll update the comment to be more detailed.

Reply:
There's got to be some way to using bounded polymorphism or some other way of specifying type variance to handle this. However, that requires experimentation, so we can do it separately.